core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1220,6 +1221,24 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleDeleteGroupsRequest(request: RequestChannel.Request): Unit = {
+    val deleteGroupsRequest = request.body[DeleteGroupsRequest]
+    var groups = deleteGroupsRequest.groups.asScala.toSet
+    var unauthorizedGroupsDeletionResult: Map[String, Errors] = Map()
+
+    groups.foreach { group =>
+      if (!authorize(request.session, Delete, new Resource(Group, group))) {
+        unauthorizedGroupsDeletionResult += (group -> Errors.GROUP_AUTHORIZATION_FAILED)
+        groups -= group
```

Comment:
Perhaps we can `partition` to split the incoming group list into the authorized and unauthorized groups.